Ios 5590 algorand make unit tests

skibinalexander opened the pull request.

skibinalexander committed:
IOS-5589 Make send transaction implementation

skibinalexander committed:
IOS-5590 Add transaction unit tests for algorand

skibinalexander committed:
IOS-5589 Fix transaction params

skibinalexander committed:
IOS-5589 Refactor transaction builder

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5589-test-signed-transaction' into IOS-5590-algorand-make-unit-tests

# Conflicts:
#	BlockchainSdk/WalletManagers/Algorand/AlgorandTransactionBuilder.swift

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5591-algorand-integration' into IOS-5590-algorand-make-unit-tests

# Conflicts:
#	BlockchainSdk.xcodeproj/project.pbxproj
#	BlockchainSdk/Common/Factories/ExternalLinkProviderFactory.swift
#	BlockchainSdk/Common/WalletAssembly/AlgorandWalletAssembly.swift
#	BlockchainSdk/WalletManagers/Algorand/AlgorandTransactionBuilder.swift
#	BlockchainSdk/WalletManagers/Algorand/AlgorandWalletManager.swift

skibinalexander committed:
IOS-5590 Make vector tests

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5589-make-transaction-builder-algorand' into IOS-5590-algorand-make-unit-tests

# Conflicts:
#	BlockchainSdk.xcodeproj/project.pbxproj
#	BlockchainSdk/BlockchainService/Algorand/AlgorandNetworkService.swift
#	BlockchainSdk/BlockchainService/Algorand/DTO/AlgorandAccountStatus.swift
#	BlockchainSdk/BlockchainService/Algorand/DTO/AlgorandResponse+Transaction.swift
#	BlockchainSdk/BlockchainService/Algorand/Provider/AlgorandNetworkProvider.swift
#	BlockchainSdk/BlockchainService/Algorand/Provider/AlgorandProviderTarget.swift
#	BlockchainSdk/Common/WalletAssembly/AlgorandWalletAssembly.swift
#	BlockchainSdk/WalletManagers/Algorand/AlgorandTransactionBuilder.swift
#	BlockchainSdk/WalletManagers/Algorand/AlgorandWalletManager.swift
#	BlockchainSdk/WalletManagers/Algorand/Domain/AlgorandTransactionParams.swift

skibinalexander committed:
IOS-5590 Fix tests

skibinalexander committed:
IOS-5590 Fix algorand tests

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5588-make-algorand-network-layer' into IOS-5590-algorand-make-unit-tests

# Conflicts:
#	BlockchainSdk.xcodeproj/project.pbxproj

skibinalexander committed:
IOS-5590 Refactor

skibinalexander committed:
IOS-5590 Refactor

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5588-make-algorand-network-layer' into IOS-5590-algorand-make-unit-tests

m3g0byt3 commented:
А этот пулл wip или нет?
И кажется тут с таргет веткой чот не то, слишком много изменений

skibinalexander committed:
IOS-5590 Fix test vectors

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5588-make-algorand-network-layer' into IOS-5590-algorand-make-unit-tests

# Conflicts:
#	BlockchainSdk/Common/Blockchain.swift
#	BlockchainSdkTests/Common/AddressTests.swift

skibinalexander commented:
Смотреть после: 

https://github.com/tangem/blockchain-sdk-swift/pull/516

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5588-make-algorand-network-layer' into IOS-5590-algorand-make-unit-tests

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5588-make-algorand-network-layer' into IOS-5590-algorand-make-unit-tests

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5588-make-algorand-network-layer' into IOS-5590-algorand-make-unit-tests

skibinalexander committed:
IOS-5590 Merge branch 'IOS-5591-algorand-integration' into IOS-5590-algorand-make-unit-tests

skibinalexander committed:
IOS-5590 Fix by merge

skibinalexander commented on BlockchainSdkTests/Common/AddressTests.swift:
```diff
@@ -1031,5 +1031,22 @@ class AddressesTests: XCTestCase {
         XCTAssertTrue(validator.validate(ethAddr))
         XCTAssertTrue(validator.validate(xdcAddr))
+    
+    func testAlgorandAddressValidation() throws {
```
> на генерацию адреса надо еще добавить ао аналогии с сетями выше

Добавлено

skibinalexander replied:
> на генерацию адреса надо еще добавить ао аналогии с сетями выше

Тестовые данные взял из wallet-core тестов

tureck1y replied:
забыл про генерацию с другими кривыми ( должна быть ошибка )

skibinalexander replied:
Да , не обратил внимание :( добавлю

skibinalexander committed:
IOS-5590 Add throws secp keys for tests

m3g0byt3 commented on BlockchainSdkTests/Algorand/AlgorandTests.swift:
```diff
@@ -0,0 +1,97 @@
+final class AlgorandTests: XCTestCase {
+    private let blockchain: BlockchainSdk.Blockchain = .algorand(curve: .ed25519_slip0010, testnet: false)
+    private let coinType: CoinType = .algorand
+    private let privateKeyData = Data(hexString: "d5b43d706ef0cb641081d45a2ec213b5d8281f439f2425d1af54e2afdaabf55b")
```
Это приватник из траста, почему не использовать тестовый сид "tiny escape drive pupil flavor endless love walk gadget match filter luxury" как в остальных тестах?

Сгенерив из него приватник на https://iancoleman.io/

m3g0byt3 commented on BlockchainSdkTests/Algorand/AlgorandTests.swift:
```diff
@@ -0,0 +1,97 @@
+    
+    // MARK: - Impementation
+    
+    func testCoreExample() throws {
```
Кажется этот тест - это копипаста из траста, разве он имеет смысл тут?

skibinalexander replied:
Скорее нет чем да, но я бы оставил для корректной валидации билда транзакции

skibinalexander replied:
> Кажется этот тест - это копипаста из траста, разве он имеет смысл тут?

Окей уберу

m3g0byt3 commented on BlockchainSdkTests/Algorand/AlgorandTests.swift:
```diff
@@ -0,0 +1,97 @@
+        XCTAssertEqual(output.encoded.hexString.lowercased(), "82a3736967c440baa00062adcdcb5875e4435cdc6885d26bfe5308ab17983c0fda790b7103051fcb111554e5badfc0ac7edf7e1223a434342a9eeed5cdb047690827325051560ba374786e8aa3616d74cf000000e8d4a51000a3666565ce00040358a26676ce001d9167a367656eac6d61696e6e65742d76312e30a26768c420c061c4d8fc1dbdded2d7604be4568e3f6d041987ac37bde4b620b5ab39248adfa26c76ce001d954f ...
+    }
+    
+    func testTransactionBuilder() throws {
```
А это какая-то реальная отправленная транза?
Есть ссылка на нее в тестнете?

Например траст тестит на https://algoexplorer.io/tx/FFLUH4QKZHG744RIQ2AZNWZUSIIH262KZ4MEWSY4RXMWN5NMOOJA

Тест транзакций имеет смысл делать, сравнивая с реальной эталон транзе в тестнет (сделанной например через SDK блокчейна, желательно с общим тестовым сидом "tiny escape drive pupil flavor endless love walk gadget match filter luxury")

И затем уже воспроизводить в своих юнитах транзу с идентичными параметрами

skibinalexander replied:
Я брал хеш с реальной транзакции которая встала в блокчейн, в целом думаю ок же

skibinalexander replied:
> А это какая-то реальная отправленная транза? Есть ссылка на нее в тестнете?
> 
> Например траст тестит на https://algoexplorer.io/tx/FFLUH4QKZHG744RIQ2AZNWZUSIIH262KZ4MEWSY4RXMWN5NMOOJA
> 
> Тест транзакций имеет смысл делать, сравнивая с реальной эталон транзе в тестнет (сделанной например через SDK блокчейна, желательно с общим тестовым сидом "tiny escape drive pupil flavor endless love walk gadget match filter luxury")
> 
> И затем уже воспроизводить в своих юнитах транзу с идентичными параметрами

Сделаю на транзакции из тестовой мнемоники

m3g0byt3 commented on BlockchainSdkTests/Common/AddressTests.swift:
```diff
@@ -1032,4 +1032,42 @@ class AddressesTests: XCTestCase {
+        let addressServiceFactory = AddressServiceFactory(blockchain: .algorand(curve: .ed25519_slip0010, testnet: false))
+        let addressService = addressServiceFactory.makeAddressService()
+        
+        let privateKey = Data(hexString: "a6c4394041e64fe93d889386d7922af1b9a87f12e433762759608e61434d6cf7")
```
И тут тоже вопрос про приватник, если сделать его из тестового сида - это позволит проверить корректность адреса в любом стороннем воллете (траст, официальный, леджер) и убедиться что все совпадает

skibinalexander replied:
Я бы не стал , проверка с нашими ключами , совпадает же в векторе, а здесь мы точно знаем тестовый набор с wallet-core

skibinalexander committed:
IOS-5590 Remove wallet-core tests

skibinalexander committed:
IOS-5590 Fix review comments

skibinalexander committed:
IOS-5590 Fix review comments

m3g0byt3 commented:
В пуле видны какие-то коммиты из IOS-5589, но в чейнджах их нету. Странно, мб гитхаб тупит ¯\_(ツ)_/¯

skibinalexander commented:
> В пуле видны какие-то коммиты из IOS-5589, но в чейнджах их нету. Странно, мб гитхаб тупит ¯_(ツ)_/¯

Вроде норм, я каскадно подтягивал изменения
